testing/test_main.py

```diff
@@ -262,3 +262,27 @@ def test(fix):
             "* 1 passed in *",
         ]
     )
+
+
+def test_very_long_cmdline_arg(pytester: Pytester) -> None:
```

Comment:
@iandonnelly I'm having trouble reproducing the original problem, this test is passing on all platforms.

Reply:
its possibly related to available file-systems/path types (i believe the github runners always use unc/long paths)

would it break when using a path thats not supported on all platforms (i believe >64k)

Reply:
Hmm thanks for the hint, trying with a very large path (>100k)

Reply:
Btw anybody on linux can test this please?

```python 
>>> import errno
>>> errno.ENAMETOOLONG
38
```

Reply:
Hmm no good using `>100k` filenames. I guess we can rely on patching then to ensure the behavior...